packages/eslint-plugin-formatjs/rules/no-multiple-whitespaces.ts

```diff
@@ -15,20 +50,14 @@ function checkNode(context: Rule.RuleContext, node: TSESTree.Node) {
     if (!defaultMessage || !messageNode) {
       continue
     }
-    if (MULTIPLE_SPACES.test(defaultMessage)) {
-      const reportObject: Parameters<typeof context['report']>[0] = {
+
+    try {
+      verifyAst(parse(defaultMessage))
+    } catch (e) {
+      context.report({
         node: messageNode as any,
-        message: 'Multiple consecutive whitespaces are not allowed',
-      }
-      if (messageNode.type === 'Literal' && messageNode.raw) {
-        reportObject.fix = function (fixer) {
-          return fixer.replaceText(
-            messageNode as any,
-            messageNode.raw!.replace(MULTIPLE_SPACES, ' ')
```

Comment:
hmm we should still be able to fix this though right? just trimming spaces in `literal` nodes

Reply:
@longlho The tricky part is how to replaceText with proper escaping (e.g. for newlines, quotes). I guess I am just being lazy. Do you know of any utilities to do this?

Reply:
I'm not sure I understand, can you provide an example?